lib/blocks.php

```diff
@@ -504,3 +504,94 @@ function gutenberg_migrate_old_typography_shape( $metadata ) {
 if ( ! function_exists( 'wp_migrate_old_typography_shape' ) ) {
 	add_filter( 'block_type_metadata', 'gutenberg_migrate_old_typography_shape' );
 }
+
+/**
+ * Enqueue a stylesheet for a specific block.
+ *
+ * If the theme has opted-in to separate-styles loading,
+ * then the stylesheet will be enqueued on-render,
+ * otherwise when the block inits.
+ *
+ * @param string $block_name The block-name, including namespace.
+ * @param array  $args       An array of arguments [handle,src,deps,ver,media].
+ * @param bool   $register   Whether the stylesheet should be registered.
+ *                           If set to false, then assume it has already been registered,
+ *                           and only enqueue it.
+ *
+ * @return void
+ */
+function gutenberg_enqueue_block_style( $block_name, $args, $register = true ) {
+	$args = wp_parse_args(
+		$args,
+		array(
+			'handle' => '',
+			'src'    => '',
+			'deps'   => array(),
+			'ver'    => false,
+			'media'  => 'all',
+		)
+	);
+
+	/**
+	 * Callback function to register and enqueue styles.
+	 *
+	 * @param string $content When the callback is used for the render_block filter,
+	 *                        the content needs to be returned so the function parameter
+	 *                        is to ensure the content exists.
+	 *
+	 * @return string
+	 */
+	$callback = function( $content ) use ( $args, $register ) {
+		// Register the stylesheet.
+		if ( $register ) {
+			wp_register_style( $args['handle'], $args['src'], $args['deps'], $args['ver'], $args['media'] );
+		}
+
+		// Add `path` data if provided.
+		if ( isset( $args['path'] ) ) {
+			wp_style_add_data( $args['handle'], 'path', $args['path'] );
+		}
```

Comment:
Looking at `register_block_style_handle` in WP core that is quite similar but for the `block.json`:

https://github.com/WordPress/wordpress-develop/blob/bb389cda941be933ced414f5d9406c170e228f01/src/wp-includes/blocks.php#L129

It looks like RTL support is missing here. Unless we assume that it has to be handled as a separate function call.

Reply:
I added RTL styles just now, in https://github.com/WordPress/gutenberg/pull/32510/commits/b2b5a4c216d864ea3f116e6ce36dd00200c480f7 but I believe we can only add them if path` is defined? :thinking: 
Considering these will be getting added from plugins & themes I can't think of a reliable way to get them without checking the path and if a file exists...

Reply:
I think you are correct. It's a temporary approach so I would assume it's fine to handle it this way.